docs/reference/api/rtos/EventQueue.md

```diff
@@ -1,11 +1,122 @@
+2. In the context of the event loop's thread function when a falling edge is detected on `SW2` (`fall_handler`). You can use the `fall_handler` function as an argument to `queue.event()` to specify that `fall_handler` runs in user context instead of interrupt context.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/events_ex_1/)](https://os.mbed.com/teams/mbed_example/code/events_ex_1/file/6ae734681f16/main.cpp)
+
+### EventQueue example: posting events to the queue
+
+The code below demonstrates queueing functions to be called after a delay and queueing functions to be called periodically.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/events_ex_2/)](https://os.mbed.com/teams/mbed_example/code/events_ex_2/file/488fe91e2e80/main.cpp)
+
+### EventQueue example: chaining events from more than one queue
+
+Event queues easily align with module boundaries, where event dispatch can implicitly synchronize internal state. Multiple modules can use independent event queues but still be composed through the `EventQueue::chain` function.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/events_ex_3/)](https://os.mbed.com/teams/mbed_example/code/events_ex_3/file/fca134a32b61/main.cpp)
+
+### Shared event example: deferring from interrupt context
+
+Like the previous example, this defers from interrupt to an event queue thread. However, rather than creating its own thread, it uses the shared event queue – potentially sharing it with other system components and saving RAM.
+
+As the event queue is shared, you should limit the execution time of your event functions to avoid delaying other users’ events excessively.
+
+```
+#include "mbed.h"
+#include "mbed_events.h"
+ 
+DigitalOut led1(LED1);
+InterruptIn sw(SW2);
+ 
+void rise_handler(void) {
+    // Toggle LED
+    led1 = !led1;
+}
+ 
+void fall_handler(void) {
+    printf("fall_handler in context %p\r\n", Thread::gettid());
+    // Toggle LED
+    led1 = !led1;
+}
+ 
+int main() {
+    // Request the shared queue
+    EventQueue *queue = mbed_event_queue();
+    printf("Starting in context %p\r\n", Thread::gettid());
+    // The 'rise' handler will execute in IRQ context
+    sw.rise(rise_handler);
+    // The 'fall' handler will execute in the context of the shared queue thread
+    sw.fall(queue->event(fall_handler));
+}
+```
+
+### Shared event example: running the shared queue from main
+
+To further save RAM, if you have no other work to do in your main function after initialization, you can dispatch the global event queue from there, avoiding the need to create a separate dispatch thread.
+
+To do this, set the `mbed_app.json` configuration option `events.shared-dispatch-from-application` to true, and add a dispatch call to main, as in this example. (The prints now show the same context for startup and `fall_handler`).
+
+```
+#include "mbed.h"
```

Comment:
@deepikabhavnani Could you please help me add this example to the teams repo, so it's not hard-coded into the docs?